hms-app/src/components/admin-tools/UserForm.tsx

```diff
@@ -0,0 +1,197 @@
+      ...prevUserText,
+      [event.target.name]: event.target.value,
+    }))
+  }
+
+  function editThisUser(event: React.MouseEvent<HTMLButtonElement>) {
+    event.preventDefault()
+    showNotification({
+      id: 'user-load',
+      loading: true,
+      title: 'User is uploading',
+      message: 'this can take a second',
+      autoClose: false,
+      disallowClose: true,
+    })
+    editUser(contextID, user, skills).then((r) =>
+      setTimeout(() => {
+        updateNotification({
+          id: 'user-load',
+          color: 'teal',
+          title: 'User was added',
+          message: 'Notification will close in 2 seconds',
+          icon: <CheckIcon />,
+          autoClose: 2000,
+        })
+      }, 3000)
+    )
+  }
+
+  function submitIsEnabled(): boolean {
+    return !!user.firstName
+  }
+
+  const loadAvailableSkills = () => {
+    getListOfSkills().then((data) => {
+      setAvailableSkills({
+        ...availableSkills,
+        skills: data.skills,
+      })
+    })
+  }
+
+  const skillsList = availableSkills.skills.map((skill, index) => [
+    <Checkbox value={skill.id} label={skill.name} />,
+  ])
+
+  return (
+    <>
+      {isLoading && (
+        <div>
+          <h3>User details are loading...</h3>
+        </div>
+      )}
+      {!isLoading && (
+        <Card withBorder radius="md" p="md" className={classes.card}>
+          <Card.Section className={classes.section}>
+            <Textarea
+              label="First Name"
+              mt="sm"
+              required
+              placeholder="First Name"
+              maxRows={1}
+              autosize
+              onChange={handleChange}
+              name="title"
+              value={user.firstName}
+            />
+          </Card.Section>
+          <Card.Section className={classes.section}>
+            <Textarea
+              label="Last Name"
+              mt="sm"
+              placeholder="Last Name"
+              maxRows={1}
+              autosize
+              onChange={handleChange}
+              name="description"
+              value={user.lastName}
+            />
+          </Card.Section>
+          <Card.Section className={classes.section}>
+            <Textarea
+              label="E-mail"
+              mt="sm"
+              placeholder="E-mail"
+              maxRows={1}
+              autosize
+              onChange={handleChange}
+              name="description"
+              value={user.emailAddress}
+            />
+          </Card.Section>
+          <Card.Section className={classes.section}>
+            <CheckboxGroup
+              mt="sm"
+              color="gray"
+              label="Skills"
+              spacing="md"
+              onChange={setSkills}
+              required
+            >
+              {skillsList}
+            </CheckboxGroup>
+          </Card.Section>
+          <Group position="right" mt="xl">
+            <Button disabled={!submitIsEnabled()} onClick={editThisUser}>
```

Comment:
the one thing about all this though is that I still haven't implemented an editUser endpoint :P

Reply:
:O